File: backend/internship-todolg-server/internship-todolg-model/src/main/java/com/comarch/fbi/internship/todolg/model/entities/Task.java
package com.comarch.fbi.internship.todolg.model.entities;

import java.time.LocalDateTime;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import org.hibernate.validator.constraints.Length;

import lombok.Getter;
import lombok.Setter;

/**
 * Encja reprezentująca zadanie.
 */
@Entity(name = "tasks")
@Getter
@Setter
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    @Length(min = 1, max = 80)
    @Column(length = 80)
    private String title;

    private LocalDateTime startDate;

    @Length(max = 255)
    @Column(length = 80)
    private String note;

    private int priority;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "group_id")
    private Group group;

    private boolean status;
}

File: frontend/src/app/models/Task.ts
export class Task {
  id: number;
  title: string;
  startDate: Date;
  note: string;
  priority: number;
  groupId: number;
  status: boolean;

  constructor(id: number, title: string, startDate: Date,
              note: string, priority: number, groupId: number,
              status: boolean) {
    this.id = id;
    this.title = title;
    this.startDate = new Date(startDate);
    this.note = note;
    this.priority = priority;
    this.groupId = groupId;
    this.status = status;
  }
}
